include/unifex/tag_invoke.hpp

```diff
@@ -32,6 +32,30 @@ struct tag_invoke_cpo {
     return tag_invoke((CPO&&)cpo, (Args &&) args...);
   }
 };
+
+template<typename CPO, typename... Args>
+using tag_invoke_result_t = decltype(tag_invoke(std::declval<CPO>(), std::declval<Args>()...));
```

Comment:
`declval` instantiates templates because it must handle the `void` case. `CPO` and `Args` can never be `void`. If you want to make this compile even faster, you can replace `declval<T>()` with `static_cast<T&&(*)()>(nullptr)()`.